control/mpc_lateral_controller/src/mpc_utils.cpp

```diff
@@ -92,7 +106,16 @@ std::pair<bool, MPCTrajectory> resampleMPCTrajectoryByDistance(
   }
 
   std::vector<double> output_arclength;
-  for (double s = 0; s < input_arclength.back(); s += resample_interval_dist) {
+  for (double s = std::clamp(
+         input_arclength.at(nearest_seg_idx) + ego_offset_to_segment, 0.0,
+         input_arclength.back() - 1e-6);
+       0 <= s; s -= resample_interval_dist) {
+    output_arclength.push_back(s);
+  }
+  std::reverse(output_arclength.begin(), output_arclength.end());
+  for (double s = std::max(input_arclength.at(nearest_seg_idx) + ego_offset_to_segment, 0.0) +
```

Comment:
is `input_arclength.at(nearest_seg_idx) + ego_offset_to_segment` resampled twice?

Reply:
No, this case uses `double s = std::max(input_arclength.at(nearest_seg_idx) + ego_offset_to_segment, 0.0) + "resample_interval_dist"`.